### PR TITLE
Implement Python 3.12 compatible loader. Fixes #130.

### LINKED ISSUE #130: PyOtherSideQtRCImporter broken on Python3.12
PyOtherSideQtRCImporter needs to implement find_spec since find_module is no longer supported. https://docs.python.org/3.11/library/importlib.html#importlib.abc.MetaPathFinder.find_module shows it is deprecated since 3.4 (the link shows the 3.11 docs) and it was removed in 3.12.

### PR DESCRIPTION
**Pull request by apollo13**

### CODE REVIEW

File: src/qrc_importer.py [hunk 1]
```diff
@@ -16,11 +16,33 @@
...
 import pyotherside
 
-from importlib import abc
+
+class PyOtherSideQtRCLoader(abc.Loader):
+    def __init__(self, filepath):
+        self.filepath = filepath
+
+    def create_module(self, spec):
+        return
```
Comment (thp) [inline src/qrc_importer.py:30]: Make it more explicit that this returns `None` as per [the docs](https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.create_module):

```suggestion
        return None
```

File: src/qrc_importer.py [hunk 2]
```diff
@@ -16,11 +16,33 @@
...
+        exec(code, module.__dict__)
+
 
 class PyOtherSideQtRCImporter(abc.MetaPathFinder, abc.SourceLoader):
+    def find_spec(self, fullname, path, target=None):
+        if path is None:
+            fname = self.get_filename(fullname)
+            if fname:
+                return spec_from_loader(fullname, PyOtherSideQtRCLoader(fname))
+        return
```
Comment (thp) [inline src/qrc_importer.py:44]: Also here, `python3 -m this | grep Explicit`:

```suggestion
        return None
```

**Review by thp — CHANGES_REQUESTED** [review]
Thanks! Some comments inline plus these:

- Add a changelog entry in `Version UNRELEASED` section in `docs/index.rst` for this
- In `.github/workflows/build.yaml`, change `python@3.8` to `python@3.12`
- As some of these loader features were only introduced in a Python version >= 3.3, we should bump the minimum version in `README.md` to Python 3.8, as everything before that [is end-of-life](https://devguide.python.org/versions/) (+ add to the changelog)

**Comment by apollo13** [comment]
@thp Is there anything you miss here?

**Review by thp — APPROVED** [review]
Nice, thanks!

**Comment by uaqben** [comment]
@thp you plan to tag a release including this fix?  fixes [yubikey-manager-qt](https://github.com/Yubico/yubikey-manager-qt/issues/357)

**Comment by thp** [comment]
> @thp you plan to tag a release including this fix? fixes [yubikey-manager-qt](https://github.com/Yubico/yubikey-manager-qt/issues/357) 😊

PyOtherSide 1.6.1 is now released.

**Pull request merged**